### PR TITLE
Read polygons from file

### PR DESCRIPTION
**Pull request by mforets**
This is code we had from https://github.com/JuliaReach/ReachabilityBenchmarks/blob/master/src/plot_GEN.jl , but it is useful quite frequently, eg. to read files generated by other tools.

### CODE REVIEW

File: src/Utils/helper_functions.jl [hunk 1]
```diff
@@ -450,3 +450,68 @@ end
...
+0.99 0.99
+1.01 0.99
+
+0.908463 1.31047
+0.873089 1.31047
+0.873089 1.28452
+0.908463 1.28452
+
+
+```
```
Comment (schillic) [inline src/Utils/helper_functions.jl:485]: Can we remove one of the empty lines?

File: src/Utils/helper_functions.jl [hunk 2]
```diff
@@ -450,3 +450,68 @@ end
...
+This is parsed as
+
+```julia
+2-element Array{VPolygon{Float64,Array{Float64,1}},1}:
+ VPolygon{Float64,Array{Float64,1}}([[1.01, 1.01], [0.99, 1.01], [0.99, 0.99], [1.01, 0.99]])
+ VPolygon{Float64,Array{Float64,1}}([[0.908463, 1.31047], [0.873089, 1.31047], [0.873089, 1.28452], [0.908463, 1.28452]])
+```
+
+The input file should end with at least one empty line before the end of files
+(this is to detect the last polygon).
```
Comment (schillic) [inline src/Utils/helper_functions.jl:495]:
```suggestion
```

File: src/Utils/helper_functions.jl [hunk 3]
```diff
@@ -450,3 +450,68 @@ end
...
+- `filename` -- path of the file containing the polygons
+
+### Output
+
+A list of polygons in vertex representation.
+
+### Notes
+
+The `x` and `y` coordinates of each vertex should be separated by an empty space,
+and polygons are separated by empty lines. For example:
```
Comment (schillic) [inline src/Utils/helper_functions.jl:471]:
```suggestion
and polygons are separated by empty lines (even the last polygon). For example:
```

File: src/Utils/helper_functions.jl [hunk 4]
```diff
@@ -450,3 +450,68 @@ end
...
+        for line in eachline(f)
+          if !isempty(line)
+              push!(Mi, map(x -> parse(Float64, x), split(line)))
+              new_polygon = true
+          elseif isempty(line) && new_polygon
+              push!(P, VPolygon(Mi))
+              Mi = Vector{Vector{Float64}}()
+              new_polygon = false
+          end
+        end
```
Comment (schillic) [inline src/Utils/helper_functions.jl:514]: The indentation needs to be fixed here.

**Review by schillic — APPROVED** [review]
I do not think that this format is restricted to (2D) polygons. Should we generally return `VPolytope`s?

File: src/Utils/helper_functions.jl (hunk 1 above)
Comment (mforets) [reply src/Utils/helper_functions.jl:485]: hmm im 99% sure it needs both empty lines to work properly.

(i didn't want to spend a lot of time adapting this function, but feel free to improve it).

File: src/Utils/helper_functions.jl (hunk 4 above)
Comment (mforets) [reply src/Utils/helper_functions.jl:514]: thanks ,i'll fix it

Comment (mforets) [reply src/Utils/helper_functions.jl:511]: thanks ,i'll fix it

**Comment by mforets** [comment]
> I do not think that this format is restricted to (2D) polygons. Should we generally return VPolytopes?

good idea. i'll create a follow-up issue.

**Comment by mforets** [comment]
(error is unrelated)

**Pull request merged**